### DISCUSSION TITLE
Qml module `RESOURCE_PREFIX` configuration

### DISCUSSION THREAD

**Discussion by Billyzou0741326** [KDAB/cxx-qt#967] [Q&A] [answered]
As per the Qt (v6.7) docs, the `RESOURCE_PREFIX` for `qt_add_qml_module` defaults to `/qt/qml/` which is the default import path of the QML engine. And (if I'm understanding corerctly) it seems that cxx-qt also generates the qml module under that default resource prefix.

As per the cxx-qt docs on `QmlModule` (for `CxxQtBuilder::qml_module`:

> The QmlModule struct’s qml_files are registered with the [Qt Resource System](https://doc.qt.io/qt-6/resources.html) in the [default QML import path](https://doc.qt.io/qt-6/qtqml-syntax-imports.html#qml-import-path) qrc:/qt/qml/uri/of/module/. Additional resources such as images can be added to the Qt resources for the QML module by specifying the qrc_files field.

So far so good.

But now, let's say I'm working on a project where all of its existing modules have their `RESOURCE_PREFIX` set to, say, `/com.example.app/imports` instead of the default `/qt/qml` (this is done in all the `CMakeLists.txt`'s `qt_add_qml_module`). 

How would I configure cxx-qt to generate its resources under the `/com.example.app/imports` prefix rather than the default `/qt/qml` prefix? Would failing to set the `RESOURCE_PREFIX` result in a "Module not found" error when trying to import a module defined on the rust side into qml?

**Comment by LeonMatthesKDAB** [comment, marked as answer]
Hi @Billyzou0741326 , thank you for your interest in CXX-Qt and for contributing this discussion.
 
Currently we do not have a way to change the `RESOURCE_PREFIX` for qml modules generated by CXX-Qt.
Because `/qt/qml/` is in the import path by default, all modules generated by CXX-Qt should already be available from QML, so you should not need to change the `RESOURCE_PREFIX`.
We also make sure that your qml plugin is registered with QML automatically through our build system (and maybe a `init` method in the future, see: #964 ). This is very similar to what `qt_add_qml_module` does and should work out of the box.

If you have another reason for changing the default `RESOURCE_PREFIX`, please let us know :)

Also, if you're using CMake, any existing calls to `qt_add_qml_module` should still work as they did before.

I hope that answers your question.
